### PR TITLE
Login api

### PR DESCRIPTION
**Pull request by PallaviMP34**
Api integration

### CODE REVIEW

File: diagramming-tool/src/App.test.js
Comment (AnushiyaM11) [inline diagramming-tool/src/App.test.js:1]: Please remove the commented lines of code

File: diagramming-tool/src/Component/Login/LoginPage.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          password: password,
+        }),
+      });
+
+      if (!response.ok) {
+        const data = await response.json();
+        throw new Error(data.error ); 
+      }
+
+      console.log('Login successful');
```
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Login/LoginPage.jsx:41]: Remove console.log if not necessary

File: diagramming-tool/src/Component/Login/LoginPage.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    if (!password.trim()) {
+      setError('Please enter your password.');
+      return;
+    }
+
+    setError('');
+
+    try {
+      const response = await fetch('http://localhost:8080/api/diagrammingtool/login', {
```
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Login/LoginPage.jsx:25]: API calls should be in a separate folder not within the component

File: diagramming-tool/src/Component/Login/intersectionObserverMock.js
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Login/intersectionObserverMock.js:1]: Move the file to a different folder

File: diagramming-tool/src/Component/Login/jest.config.js
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Login/jest.config.js:1]: Move the file to root folder

File: diagramming-tool/src/App.test.js [hunk 3]
```diff
@@ -1,8 +1,68 @@
-import { render, screen } from '@testing-library/react';
+/* eslint-disable testing-library/prefer-presence-queries */
```
Comment (Sharz-dev) [inline diagramming-tool/src/App.test.js:1]: Please Remove the commands in the code

File: diagramming-tool/src/Component/Login/Login.test.js
Comment (Sharz-dev) [inline diagramming-tool/src/Component/Login/Login.test.js:1]: Please remove the commented lines of code

**Pull request merged**